swift-sdk/Internal/IterableAPIMobileFrameworkDetector.swift

```diff
@@ -0,0 +1,81 @@
+import Foundation
+
+final class IterableAPIMobileFrameworkDetector {
+    private struct FrameworkClasses {
+        static let flutter = [
+            "FlutterViewController",
+            "GeneratedPluginRegistrant",
+            "FlutterEngine",
+            "FlutterPluginRegistry"
+        ]
+        
+        static let reactNative = [
+            "RCTBridge",
+            "RCTRootView",
+            "RCTBundleURLProvider",
+            "RCTEventEmitter"
+        ]
+    }
+    
+    private struct BundleIdentifiers {
+        static let executableKey = "CFBundleExecutable"
+        static let flutterTargetKey = "FlutterDeploymentTarget"
+        static let reactNativeProviderKey = "RNBundleURLProvider"
+        static let flutterExecutableName = "Runner"
+    }
+    
+    private static var cachedFrameworkType: IterableAPIMobileFrameworkType = {
+        detectFramework()
+    }()
+    
+    static func detectFramework() -> IterableAPIMobileFrameworkType {
+        let bundle = Bundle.main
+        
+        // Helper function to check for framework classes
+        func hasFrameworkClasses(_ classNames: [String]) -> Bool {
+            guard !classNames.isEmpty else { return false }
+            return classNames.contains { className in
+                guard IterableUtil.isNotNullOrEmpty(string: className) else { return false }
+                return bundle.classNamed(className) != nil
+            }
+        }
+        
+        // Safely check frameworks
+        let hasFlutter = hasFrameworkClasses(FrameworkClasses.flutter)
+        let hasReactNative = hasFrameworkClasses(FrameworkClasses.reactNative)
+        
+        switch (hasFlutter, hasReactNative) {
```

Comment:
We could probably refactor this later in the case where we have more than 2 frameworks.